frmts/vrt/gdaltileindexdataset.cpp

```diff
@@ -0,0 +1,3642 @@
+                                               MD_RESAMPLING};
+
+constexpr const char *const MD_BAND_OFFSET = "OFFSET";
+constexpr const char *const MD_BAND_SCALE = "SCALE";
+constexpr const char *const MD_BAND_UNITTYPE = "UNITTYPE";
+constexpr const char *const apszReservedBandItems[] = {
+    MD_BAND_OFFSET, MD_BAND_SCALE, MD_BAND_UNITTYPE};
+
+constexpr const char *GTI_XML_BANDCOUNT = "BandCount";
+constexpr const char *GTI_XML_DATATYPE = "DataType";
+constexpr const char *GTI_XML_NODATAVALUE = "NoDataValue";
+constexpr const char *GTI_XML_COLORINTERP = "ColorInterp";
+constexpr const char *GTI_XML_LOCATIONFIELD = "LocationField";
+constexpr const char *GTI_XML_SORTFIELD = "SortField";
+constexpr const char *GTI_XML_SORTFIELDASC = "SortFieldAsc";
+constexpr const char *GTI_XML_MASKBAND = "MaskBand";
+constexpr const char *GTI_XML_OVERVIEW_ELEMENT = "Overview";
+constexpr const char *GTI_XML_OVERVIEW_DATASET = "Dataset";
+constexpr const char *GTI_XML_OVERVIEW_LAYER = "Layer";
+constexpr const char *GTI_XML_OVERVIEW_FACTOR = "Factor";
+
+constexpr const char *GTI_XML_BAND_ELEMENT = "Band";
+constexpr const char *GTI_XML_BAND_NUMBER = "band";
+constexpr const char *GTI_XML_BAND_DATATYPE = "dataType";
+constexpr const char *GTI_XML_BAND_DESCRIPTION = "Description";
+constexpr const char *GTI_XML_BAND_OFFSET = "Offset";
+constexpr const char *GTI_XML_BAND_SCALE = "Scale";
+constexpr const char *GTI_XML_BAND_NODATAVALUE = "NoDataValue";
+constexpr const char *GTI_XML_BAND_UNITTYPE = "UnitType";
+constexpr const char *GTI_XML_BAND_COLORINTERP = "ColorInterp";
+constexpr const char *GTI_XML_CATEGORYNAMES = "CategoryNames";
+constexpr const char *GTI_XML_COLORTABLE = "ColorTable";
+constexpr const char *GTI_XML_RAT = "GDALRasterAttributeTable";
+
+/************************************************************************/
+/*                           ENDS_WITH_CI()                             */
+/************************************************************************/
+
+static inline bool ENDS_WITH_CI(const char *a, const char *b)
+{
+    return strlen(a) >= strlen(b) && EQUAL(a + strlen(a) - strlen(b), b);
+}
+
+/************************************************************************/
+/*                       GDALTileIndexDataset                           */
+/************************************************************************/
+
+class GDALTileIndexBand;
+class GDALTileIndexDataset final : public GDALPamDataset
+{
+    friend class GDALTileIndexBand;
+
+    CPLXMLTreeCloser m_psXMLTree{nullptr};
+    bool m_bXMLUpdatable = false;
+    bool m_bXMLModified = false;
+    const std::string m_osUniqueHandle;
+    std::unique_ptr<GDALDataset> m_poVectorDS{};
+    OGRLayer *m_poLayer = nullptr;
+    std::array<double, 6> m_adfGeoTransform{0, 0, 0, 0, 0, 0};
+    int m_nLocationFieldIndex = -1;
+    OGRSpatialReference m_oSRS{};
+    lru11::Cache<std::string, std::shared_ptr<GDALDataset>> m_oMapSharedSources{
```

Comment:
I'm wondering if the cache size here should be user configurable.

Reply:
I don't see a need for the cache size to be user configurable. At least at that level

    //! Cache from dataset name to dataset handle.
    //! Note that the dataset objects are ultimately GDALProxyPoolDataset,
    //! and that the GDALProxyPoolDataset limits the number of simultaneously
    //! opened real datasets (controlled by GDAL_MAX_DATASET_POOL_SIZE). Hence 500 is not too big.